client/src/context/HomeContext/index.js

```diff
@@ -53,6 +56,8 @@ const useHome = () => {
   const fetchMessages = useRef(fetchMessagesApi(setChannelsMap.current));
   const newMessage = useRef(newMessageHandler(dispatch));
   const setFirstUserStatus = useRef(setFirstUserStatusHandler(dispatch));
+  const setUserOffline = useRef(setUserOfflineHandler(dispatch));
```

Comment:
Why useRef and not useCallback?

Reply:
Well I was not sure about `useCallback` here. I am going through its benefits. And I will start using it, where ever it can be used.
Thanks.